server/ConnectionHub.js

```diff
@@ -0,0 +1,42 @@
+let connectionHubInstance = null;
+
+class ConnectionHub {
+  static Instance(){
+    if(connectionHubInstance === null){
+      connectionHubInstance = new ConnectionHub();
+    }
+
+    return connectionHubInstance;
+  }
+
+  constructor() {
+    this.socketById = new Map();
+  }
+
+  registerConnection(socket) {
+    console.log(`Registered ${socket.id}`);
+    this.socketById[socket.id] = socket;
```

Comment:
if you're using Map() syntax, this should be this.socketById.set(socket.id, socket);
by using the a[b]=c syntax you're just modifying the underlying js object

Reply:
yeah haha i realized that when I woke up